Content.Shared/Physics/CollisionGroup.cs

```diff
@@ -23,6 +23,7 @@ public enum CollisionGroup
         GhostImpassable = 1 <<  5, // 32 Things impassible by ghosts/observers, ie blessed tiles or forcefields
         Underplating    = 1 <<  6, // 64 Things that are under plating
         Passable        = 1 <<  7, // 128 Things that are passable
+        SingularityImpassable = 1 <<  8, // 256 Objects which are specifically impassable by the singularity
```

Comment:
This is silly because it just clutters collision layers and doesn't account for the existing PreventCollideEvent that actually checks for the singulo level.

Reply:
Either PreventCollideEvent would have to prevent collisions with the map and every single entity, or it's this.
Too many bugs have occurred where singularity doesn't collide with stuff it should.

Reply:
Or does collide with stuff it shouldn't, which is why this got introduced.

Reply:
The only things it should be hard colliding with at all are fields and generators. Anything else is a bug.

Reply:
Specifically, this got introduced because the singularity was colliding with:
+ Walls
+ All types of floor (due to the new shuttle collision logic)
+ Pretty much anything (it was set to `AllMask`)

Reply:
It was set to AllMask because legacy shitcode.

It's still replacing shitcode with new shitcode. PreventCollide can already handle what singulo needs handling so update it.

Reply:
Using existing collision groups for singulo was what lead to the breakage where stuff wouldn't collide with it because the collision groups were wrong.
See the entire confusion in https://github.com/space-wizards/space-station-14/pull/3963 for details.

Reply:
Basically, either it gets collision groups to prevent unexpected interactions, or it stays at AllMask and absolutely all processing goes through PreventCollide (which does not sound processor-friendly).

Reply:
> Also getting crashes whenever singulo remotely goes near station due to 1 of several things but this also happens on master now so.
> 
> Radiation also still seems lethal, just I died in 3 seconds instead of 1.

Not getting crashes when singulo goes near station here, even on debug.

I'm not sure what's up with the radiation being instantly lethal. I suspect it used to be distance-based and now it isn't - it was always pretty lethal at close range.

Reply:
Adding an entirely new collision group is going to have more of a performance impact than running preventcollideevent.

Reply:
Not sure how adding a new collision group will have more a performance impact. Indications seem to be they're just bitmasks and aren't checked like that (if they were, there'd probably have to be an explicit amount of collision groups to fill out, or some way of tracking which ones have been used, neither of which I see here)

Reply:
FlagSerializer increments.

Having both physics and something that blindly updates every tick is dumb because you're giving it the highest performance cost possible (short of increasing the range again).

Reply:
Would it be okay if carpets and such got static collision bodies that collide with almost nothing? Because if that was allowed then the blind update every tick can be removed entirely.